osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs

```diff
@@ -225,6 +228,7 @@ private List<GroupMapping> getGroupsBy(Func<BeatmapInfo, GroupDefinition> getGro
         {
             return items.GroupBy(i => getGroup((BeatmapInfo)i.Model))
                         .OrderBy(s => s.Key.Order)
+                        .ThenBy(s => s.Key.Title)
```

Comment:
Is this implying there's groups with the same order now..?

Reply:
Yes, specifically in source grouping mode, where I want all groups (excluding unsourced maps) to be sorted lexicographically. I can't achieve that by the `Order` field because of the way grouping is implemented, so rather than rewriting it again, I've went with the cheaper option of giving all groups the same order number and let it fall back to sorting by group title.

I can see a possibility to rewrite grouping to work similar to how it's implemented in stable (e.g. [here](https://github.com/peppy/osu-stable-reference/blob/a5e5fe6ef240505d13526cf32783cad261e9bd8b/osu!/GameplayElements/Beatmaps/BeatmapTreeManager.cs#L2153-L2164)), that way order is defined by which group is added first by the code, rather than relying on `Order`/`Title`. Should I proceed with that path?